src/cli-validator/utils/fixProblems.js

```diff
@@ -0,0 +1,259 @@
+  const markForMoving = (path, parent) => {
+    const pathstr = path.slice(0, path.length - 1).join(',');
+    if (
+      !(
+        pathstr in paramsToDelete &&
+        path[path.length - 1] in paramsToDelete[pathstr]
+      )
+    ) {
+      if (pathstr in paramsToPrepend) {
+        paramsToPrepend[pathstr].push(parent[path[path.length - 1]]);
+      } else {
+        paramsToPrepend[pathstr] = [parent[path[path.length - 1]]];
+      }
+    }
+  };
+
+  // Unmark a parameter for Moving if it is already marked for deletion
+  const unmarkIfMarkedForMoving = (path, parent) => {
+    const pathstr = path.slice(0, path.length - 1).join(',');
+    const compFunc = param =>
+      JSON.stringify(param) === JSON.stringify(parent[path[path.length - 1]]);
+    if (
+      pathstr in paramsToPrepend &&
+      paramsToPrepend[pathstr].findIndex(compFunc) != -1
+    ) {
+      paramsToPrepend[pathstr].splice(
+        paramsToPrepend[pathstr].findIndex(compFunc),
+        1
+      );
+    }
+  };
+
+  // Given a path array, traverse down the path on the original JSON object
+  // Return the found object and up to three parents
+  const getObjectFromPath = path => {
+    const pathcopy = path.slice(0);
+    let obj = originalJSON;
+    let parent, pParent, ppParent;
+    while (pathcopy.length) {
+      ppParent = pParent;
+      pParent = parent;
+      parent = obj;
+      obj = obj[pathcopy.shift()];
+    }
+    return { obj, parent, pParent, ppParent };
+  };
+
+  // For each problem type and for every problem
+  types.forEach(type => {
+    each(results[type], problems => {
+      problems.forEach(problem => {
+        const message = problem.message.split(':')[0];
+        let path = problem.path;
+        // path needs to be an array to get the line number
+        if (!Array.isArray(path)) {
+          path = path.split('.');
+        }
+
+        // In the case that a path includes brackets ['path', 'to', 'array[index]'],
+        // remove the brackets and add the index inside as a new element ['path', 'to', 'array', 'index']
+        if (path[path.length - 1].includes('[')) {
+          path.push(path[path.length - 1].split(']')[0].split('[')[1]);
+          path[path.length - 2] = path[path.length - 2].split(['['])[0];
+        }
+
+        // Get object and it's parents from path
+        const { obj, parent, ppParent } = getObjectFromPath(path);
+
+        // Fix changes based on given warning or error message
+        if (message.includes('Schema must have a non-empty description')) {
+          // obj['description'] = path[path.length - 1];
+        } else if (
+          message.includes(
+            'Schema properties must have a description with content in it'
+          )
+        ) {
+          // parent['description'] = path[path.length - 2];
+        } else if (
+          message.includes('Definition was declared but never used in document')
+        ) {
+          definitionsToDelete[path.join(',')] = true;
```

Comment:
If you made `definitionsToDelete` an array, you could simply push the path (leaving it as an array) onto `definitionsToDelete` and avoid the `join` here and corresponding `split` below when you process the deletes.

Reply:
Good idea, done